excel.c

```diff
@@ -2380,12 +2401,12 @@ zend_bool php_excel_write_cell(SheetHandle sheet, BookHandle book, int row, int
 
 		case IS_STRING:
 			data_zs = Z_STR_P(data);
-			if (dtype == PHP_EXCEL_STRING) {
-				return xlSheetWriteStr(sheet, row, col, (const char*) ZSTR_VAL(data_zs), format);
-			}
 			if (Z_STRLEN_P(data) > 0 && '\'' == Z_STRVAL_P(data)[0]) {
 				return xlSheetWriteStr(sheet, row, col, (const char*) ZSTR_VAL(data_zs) + 1, format);
 			}
+			if (dtype == PHP_EXCEL_STRING) {
+				return xlSheetWriteStr(sheet, row, col, (const char*) ZSTR_VAL(data_zs), format);
+			}
```

Comment:
I've removed your code from this PR but I am still not convinced that the proposed order is the better option since it behaves different from Excel.